### PR TITLE
fix: fail UR transaction if chain IDs don't match

### PR DESCRIPTION
**Pull request by just-toby**
this may prevent a mysterious potential bug causing users to send funds to the mainnet UR address on base

from the ethers docs

transactionRequest.chainId ⇒ number | Promise< number >
The chain ID this transaction is authorized on, as specified by [EIP-155](https://eips.ethereum.org/EIPS/eip-155).
If the chain ID is 0 will disable EIP-155 and the transaction will be valid on any network. This can be dangerous and care should be taken, since it allows transactions to be replayed on networks that were possibly not intended. Intentionally-replayable transactions are also disabled by default on recent versions of Geth and require configuration to enable.

### CODE REVIEW

**Review by gnewfield — APPROVED** [review]

**Review by tinaszheng — APPROVED** [review]

**Comment by zzmp** [comment]
This should include logging so we know when it is occuring.

File: src/hooks/useUniversalRouter.ts [hunk 1]
```diff
@@ -58,6 +58,8 @@ export function useUniversalRouterSwapCallback(
         if (!chainId) throw new Error('missing chainId')
         if (!provider) throw new Error('missing provider')
         if (!trade) throw new Error('missing trade')
+        const connectedChainId = await provider.getSigner().getChainId()
+        if (chainId !== connectedChainId) throw new Error('signer chainId does not match')
```
Comment (zzmp) [inline src/hooks/useUniversalRouter.ts:62]: Will this bubble up to Sentry?

**Review by zzmp — APPROVED** [review]

**Review by JackShort — APPROVED** [review]
YEET

File: src/hooks/useUniversalRouter.ts (hunk 1 above)
Comment (just-toby) [reply src/hooks/useUniversalRouter.ts:62]: no currently we aren't logging general swap failures to sentry

**Pull request merged**